### PR TITLE
Update to adapter-react-v5 and Mui v5, add SettingsHeader (Logo) and TreeTable

### PR DESCRIPTION
**Pull request by xXBJXx**
* I have pulled up everything to the new version of the adapter-react-v5 and to mui-v5
* On the docu pages, I have also adapted all import things to the new imports
* I added a changelog and LICENSE
* updated all dependencies
* add `Logo` component and docu
* add `TreeTable` component and ducu
* add `PasswordInput` component with `show / hide` password and docu
* add `NumberImput` component with `min / max` option and docu for NumberImput

**For the docu pages I don't know if there is still a build to be done somewhere, if so please still do it** 

But there is a problem, if you try to update the socket-client to a version higher than v1.1.1 then
the adapter UI doesn't connect anymore
image
could you please have a look what the problem could be so that you can go further up in the future?
I have pulled up the version times to the 1.0.12, which still works flawlessly.
I have created an [Issue](https://github.com/ioBroker/socket-client/issues/16) in the ioBroker/socket-client for this purpose

### CODE REVIEW

File: docs/hooks/useIoBrokerState.md [hunk 1]
```diff
@@ -92,7 +92,7 @@ const MyComponent: React.FC<MyAppProps> = (props) => {
 import { useIoBrokerState } from "iobroker-react/hooks";
 
 const MyComponent: React.FC<MyAppProps> = (props) => {
-	const [, myStateAck, setMyState] = useIoBrokerState({
+	const [myState, myStateAck, setMyState] = useIoBrokerState({
```
Comment (AlCalzone) [inline docs/hooks/useIoBrokerState.md:95]: myState isn't used. No need to declare it here, or what am I missing?

File: src/lib/components/Logo.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import { useIoBrokerObject } from "../hooks/useIoBrokerObject";
+
+interface LogoProps {
+	classes?: { buttons?: string; logo?: string };
+}
+
+export const Logo: React.FC<LogoProps> = (props): JSX.Element => {
+	const { instance, namespace } = useGlobals();
+	const { showNotification } = useDialogs();
+	const [myObject, setObject] = useIoBrokerObject(
```
Comment (AlCalzone) [inline src/lib/components/Logo.tsx:14]: The 2nd array item is `extendObject`, not `setObject`. What exactly is this supposed to do?

**Review by AlCalzone — COMMENTED** [review]
First preliminary review, I'm too tired for the rest now.
I'll still need to look at the new components you added, the updated ones are mostly fine.

File: docs/hooks/useIoBrokerState.md (hunk 1 above)
Comment (xXBJXx) [reply docs/hooks/useIoBrokerState.md:95]: Yes, you are right, I overlooked that.

File: src/lib/components/Logo.tsx (hunk 2 above)
Comment (xXBJXx) [reply src/lib/components/Logo.tsx:14]: I had tried it with extendObject, but that did not have the desired effect with the `extendObject` a deep merge is made, with the `handleLoadConfig`, where setObject is used, the old config is reloaded, but it should not be merged, but overwritten or am I wrong?

**Review by AlCalzone — APPROVED** [review]

**Pull request merged**